### ISSUE TITLE
Unable to perform 'az network vnet peering create' from az cloud shell

### ISSUE THREAD

**Issue by hardjopranoto** [Azure/azure-cli#11701]
## Describe the bug

**Command Name**
`az network vnet peering create`

**Errors:**
```
400 Client Error: Bad Request for url: http://localhost:50342/oauth2/token
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 608, in execute
    raise ex
cli/core/commands/__init__.py, ln 666, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 659, in _run_job
    six.reraise(*sys.exc_info())
...
python3.6/site-packages/msrestazure/azure_active_directory.py, ln 486, in get_msi_token
    result.raise_for_status()
python3.6/site-packages/requests/models.py, ln 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:50342/oauth2/token
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az network vnet peering create -n {} -g {} --remote-vnet {} --vnet-name {} --allow-vnet-access`

## Environment Summary
```
Linux-4.15.0-1064-azure-x86_64-with-debian-stretch-sid
Python 3.6.5
Shell: bash

azure-cli 2.0.78

```

**Comment by bim-msft** [comment]
@MyronFanQiu Please take a look.

**Comment by MyronFanQiu** [comment]
@jiasli Hello. The error message is related with AAD. Could you please take a look?

**Comment by jiasli** [comment]
This is Cloud Shell issue. Could you run with `--debug` and share the output? 

To get unblocked, please run `az login` and retry the command.

**Comment by MyronFanQiu** [comment]
@hardjopranoto Do you solve this issue? If so, I will close this issue later.

**Comment by qianwens** [comment]
Close this issue since a long time no response. Please help create a new issue if you still meet the error.

**Comment by wout-scorito** [comment]
I have the same problem, see the below debug output

Command arguments: ['network', 'vnet', 'peering', 'create', '-g', 'clusterenvironment-rg-network-we', '-n', 'clustertomain', '--remote-vnet', 'company-vnet', '--vnet-name', 'company-k8s-clusterenvironment-vnet-we', '--debug']
Event: Cli.PreExecute []
Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f2ea39d9950>, <function OutputProducer.on_global_arguments at 0x7f2ea35136a8>, <function CLIQuery.on_global_arguments at 0x7f2ea32a2950>]
Event: CommandInvoker.OnPreCommandTableCreate []
Modules found from index for 'network': ['azure.cli.command_modules.natgateway', 'azure.cli.command_modules.network', 'azure.cli.command_modules.privatedns']
Loading command modules:
Name                  Load Time    Groups  Commands
natgateway                0.005         3         6
network                   0.030       126       580
privatedns                0.004        14        66
Total (3)                 0.039       143       652
These extensions are not installed and will be skipped: ['azext_ai_examples']
Loading extensions:
Name                  Load Time    Groups  Commands  Directory
Total (0)                 0.000         0         0  
Loaded 141 groups, 652 commands.
Found a match in the command table.
Raw command  : network vnet peering create
Command table: network vnet peering create
Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x7f2ea2702950>]
az_command_data_logger : command args: network vnet peering create -g {} -n {} --remote-vnet {} --vnet-name {} --debug
metadata file logging enabled - writing logs to '/home/wout/.azure/commands'.
Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x7f2ea2685378>, <function register_global_query_examples_argument.<locals>.register_query_examples at 0x7f2ea2617268>]
Event: CommandInvoker.OnPostArgumentLoad []
Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x7f2ea26172f0>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x7f2ea2617400>]
Event: CommandInvoker.OnCommandTableLoaded []
Event: CommandInvoker.OnPreParseArgs []
Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f2ea3513730>, <function CLIQuery.handle_query_parameter at 0x7f2ea32a29d8>, <function register_global_query_examples_argument.<locals>.handle_example_parameter at 0x7f2ea26171e0>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x7f2ea2617378>]
Getting management service client client_type=NetworkManagementClient
msrest.universal_http.requests : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
Adding custom headers to the client:
    'x-ms-client-request-id': 'd34022ca-5030-11eb-8a10-7085c278db69'
    'CommandName': 'network vnet peering create'
    'ParameterSetName': '-g -n --remote-vnet --vnet-name --debug'
AdalAuthentication.signed_session invoked by Track 1 SDK
Retrieving token from ADAL for resource 'https://management.core.windows.net/'
attempting to read file /home/wout/.azure/accessTokens.json as utf-8-sig
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - Authority:Performing instance discovery: ...
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - Authority:Performing static instance discovery
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - Authority:Authority validated via static instance discovery
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - TokenRequest:Getting token from cache with refresh if necessary.
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - CacheDriver:finding with query keys: {'_clientId': '...', 'userId': '...'}
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - CacheDriver:Looking for potential cache entries: {'_clientId': '...', 'userId': '...'}
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - CacheDriver:Found 4 potential entries.
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - CacheDriver:Resource specific token found.
adal-python : e418e4bd-e76d-4f13-a64e-9bf491c6091e - CacheDriver:Returning token from cache lookup, AccessTokenId: b'lWJ1lZUjgRTRIHLF5/4riBEsH6EPYU3n+xN5AYMadkA=', RefreshTokenId: b'xsryzVVUUtgJ4yLk/HMKNdCsui+WPG4d3kUSfgGOOFM='msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions/2441ebe0-6aec-48bc-ae95-8ba725867b6b/resourceGroups/clusterenvironment-rg-network-we/providers/Microsoft.Network/virtualNetworks/company-k8s-clusterenvironment-vnet-we/virtualNetworkPeerings/clustertomain?api-version=2020-07-01'
msrest.http_logger : Request method: 'PUT'
msrest.http_logger : Request headers:
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger :     'Content-Length': '554'
msrest.http_logger :     'User-Agent': 'python/3.6.10 (Linux-4.4.0-19041-Microsoft-x86_64-with-debian-buster-sid) msrest/0.6.18 msrest_azure/0.6.3 azure-mgmt-network/13.0.0 Azure-SDK-For-Python AZURECLI/2.16.0 (DEB)'
msrest.http_logger : Request body:
msrest.http_logger : {"id": "/subscriptions/2441ebe0-6aec-48bc-ae95-8ba725867b6b/resourceGroups/clusterenvironment-rg-network-we/providers/Microsoft.Network/virtualNetworks/company-k8s-clusterenvironment-vnet-we", "properties": {"allowVirtualNetworkAccess": false, "allowForwardedTraffic": false, "allowGatewayTransit": false, "useRemoteGateways": false, "remoteVirtualNetwork": {"id": "/subscriptions/2441ebe0-6aec-48bc-ae95-8ba725867b6b/resourceGroups/clusterenvironment-rg-network-we/providers/Microsoft.Network/virtualNetworks/company-vnet"}, "syncRemoteAddressSpace": false}, "name": "clustertomain"}
msrest.universal_http : Configuring redirects: allow=True, max=30
msrest.universal_http : Configuring request: timeout=100, verify=True, cert=None
msrest.universal_http : Configuring proxies: ''
msrest.universal_http : Evaluate proxies against ENV settings: True
urllib3.connectionpool : Starting new HTTPS connection (1): management.azure.com:443
urllib3.connectionpool : https://management.azure.com:443 "PUT /subscriptions/2441ebe0-6aec-48bc-ae95-8ba725867b6b/resourceGroups/clusterenvironment-rg-network-we/providers/Microsoft.Network/virtualNetworks/company-k8s-clusterenvironment-vnet-we/virtualNetworkPeerings/clustertomain?api-version=2020-07-01 HTTP/1.1" 400 366
msrest.http_logger : Response status: 400
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Content-Length': '366'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'x-ms-request-id': 'fd59cffd-74ec-4461-b829-991cbd4e5156'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'Server': 'Microsoft-HTTPAPI/2.0, Microsoft-HTTPAPI/2.0'
msrest.http_logger :     'x-ms-ratelimit-remaining-subscription-writes': '1199'
msrest.http_logger :     'x-ms-correlation-request-id': 'd538288b-9066-40a3-9bcb-34baaec5a110'
msrest.http_logger :     'x-ms-routing-request-id': 'WESTEUROPE:20210106T150701Z:d538288b-9066-40a3-9bcb-34baaec5a110'
msrest.http_logger :     'X-Content-Type-Options': 'nosniff'
msrest.http_logger :     'Date': 'Wed, 06 Jan 2021 15:07:00 GMT'
msrest.http_logger : Response content:
msrest.http_logger : {
  "error": {
    "code": "InvalidRequestFormat",
    "message": "Cannot parse the request.",
    "details": [
      {
        "code": "InvalidJson",
        "message": "Could not find member 'syncRemoteAddressSpace' on object of type 'VirtualNetworkPeeringProperties'. Path 'properties.syncRemoteAddressSpace', line 1, position 521."
      }
    ]
  }
}
msrest.exceptions : Cannot parse the request.
azure.cli.core.util.handle_exception is called with an exception:
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 654, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 717, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 710, in _run_job
    six.reraise(*sys.exc_info())
  File "/opt/az/lib/python3.6/site-packages/six.py", line 703, in reraise
    raise value
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 688, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 325, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 784, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/network/custom.py", line 6012, in create_vnet_peering
    resource_group_name, virtual_network_name, virtual_network_peering_name, peering)
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/network/v2020_07_01/operations/_virtual_network_peerings_operations.py", line 280, in create_or_update
    **operation_config
  File "/opt/az/lib/python3.6/site-packages/azure/mgmt/network/v2020_07_01/operations/_virtual_network_peerings_operations.py", line 231, in _create_or_update_initial
    raise exp
msrestazure.azure_exceptions.CloudError: Azure Error: InvalidRequestFormat
Message: Cannot parse the request.
Exception Details:
        Error Code: InvalidJson
        Message: Could not find member 'syncRemoteAddressSpace' on object of type 'VirtualNetworkPeeringProperties'. Path 'properties.syncRemoteAddressSpace', line 1, position 521.

cli.azure.cli.core.azclierror : BadRequestError: Cannot parse the request.
BadRequestError: Cannot parse the request.
Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x7f2ea2702b70>]

**Comment by yonzhan** [comment]
@msyyc @jsntcy to take a look

**Comment by jsntcy** [comment]
@wout-scorito, which CLI version did you use? if you didn't use latest one, please install [the latest version](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) and retry.

**Comment by wout-scorito** [comment]
yes, that solved the issue, i forgot to mention that here. i was using 2.15 (i think) which had a bug in it, i upgraded to the latest version which solved the issue.

thanks!
